cmd/criticize/main.go

```diff
@@ -153,10 +154,16 @@ func (l *linter) CheckPackage(pkgPath string) {
 	l.ctx.TypesInfo = &pkgInfo.Info
 	l.ctx.Package = pkgInfo.Pkg
```

Comment:
We can add `Filename` to `lint.Context` instead of `lint.context` and assign to it here.
It's read-only info that can be stored there.

Or did you had something in mind in favor of checker-local context?